aca_required_temp.py

```diff
@@ -0,0 +1,189 @@
+
+import numpy as np
+
+import agasc
+from Chandra.Time import DateTime
+from Ska.Sun import nominal_roll, pitch
+from Ska.quatutil import radec2yagzag
+from Quaternion import Quat
+import chandra_aca
+from starcheck.star_probs import t_ccd_warm_limit
+from astropy.table import Table
+
+N_ACQ_STARS = 5
+EDGE_DIST = 30
+COLD_T_CCD = -21
+WARM_T_CCD = -5
+
+ROLL_TABLE = Table.read('roll_limits.dat', format='ascii')
+
+# Save temperature calc a combination of stars
+# indexed by hash of agasc ids
+TEMP_CACHE = {}
+
+
+def get_options():
+    import argparse
+    parser = argparse.ArgumentParser(
+        description="Get required ACA temp for an attitude over a cycle")
+    parser.add_argument("ra",
+                        type=float)
+    parser.add_argument("dec",
+                        type=float)
+    parser.add_argument("--out",
+                        default="roll_temps.dat")
+    parser.add_argument("--start",
+                        default="2014-09-01")
+    parser.add_argument("--stop",
+                        default="2015-12-31")
+    opt = parser.parse_args()
+    return opt
+
+
+def get_agasc_cone(ra, dec, time=None, faint_lim=10.8):
+    cone_stars = agasc.get_agasc_cone(ra, dec, date=time)
+    cols = ['AGASC_ID', 'MAG_ACA', 'COLOR1',
+            'RA_PMCORR', 'DEC_PMCORR']
+    ok_cone_stars = cone_stars[(cone_stars['MAG_ACA'] < faint_lim)
+                               & (cone_stars['CLASS'] == 0)
+                               & (cone_stars['ASPQ1'] == 0)
+                               & (cone_stars['COLOR1'] != 0.7)][cols]
+    ok_cone_stars.sort('MAG_ACA')
+    return ok_cone_stars
+
+
+def select_fov_stars(ra, dec, roll, cone_stars):
```

Comment:
This is a fine start for just getting things rolling, but eventually we will need more accurate star selection.  From the email spec: "Identify all stars that meet the most liberal star selection requirements in SAUSAGE and pick the 8 brightest".  The point being that a field with 3 very close bright stars can suddenly go from looking great to looking horrible if you apply spoiler rules.  

I don't see a way around coding up a mini-SAUSAGE here since we are not providing statistical guidance of star catalog quality, but rather specific information about each target.

That's a long way of saying that eventually this function should be called `select stars()`.